connectors/src/lib/models.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  CreationOptional,
+  DataTypes,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+  Model,
+  Sequelize,
+} from 'sequelize';
+
+const { CONNECTORS_DATABASE_URI } = process.env;
+
+export const sequelize_conn = new Sequelize(CONNECTORS_DATABASE_URI as string, {});
+
+export class SlackConfiguration extends Model<
+  InferAttributes<SlackConfiguration>,
+  InferCreationAttributes<SlackConfiguration>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare slackTeamId: string;
+}
+
+SlackConfiguration.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    slackTeamId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    indexes: [{ fields: ['slackTeamId'] }],
+    modelName: 'slack_configurations',
+  }
+);
+
+export class Connector extends Model<InferAttributes<Connector>, InferCreationAttributes<Connector>> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare type: 'slack' | 'notion' | 'google';
+  declare nangoConnectionId: string;
+  declare dustAPIKey: string;
+  declare dustWorkspaceId: string;
+  declare dustDataSourceId: string;
+  declare slackConfigurationId: ForeignKey<SlackConfiguration['id']>;
```

Comment:
Should we do it the way around otherwise we'll have all connection types here maybe 200 in 6 months?

Reply:
Having it in this way allows to add a constraint (if type = slack, you require slackConfigurationId to be non-null).
The other way around you cannot guarantee consistency at DB level

Reply:
(We may want to add this constraint now though)

Reply:
After thinking about it one more time, I agree that having a `slack_configuration.connectorId` seems cleaner and does provide some degree of integrity.
You could have a connector that does not have any configuration, but that's also enforceable by database trigger if we wanted to (on slack_configuration delete => delete connector).
Not a big deal in my opinion.

Reply:
I'm OK either way ! I don't think DB triggers are usually a good idea though, but I can live without the DB-enforced integrity

Reply:
Lets skip the DB trigger and just have `slackConfiguration.connectorId`